src/util/draftutil.ts

```diff
@@ -314,3 +298,146 @@ export const setupPicks: (rows: number, cols: number) => any[][][] = (rows: numb
+  const errors = [];
+  if (!format?.packs) return ['Internal error in the format.'];
+  if (!format.title.trim()) errors.push('title must not be empty.');
+  if (format.packs.length === 0) errors.push('Format must have at least 1 pack.');
+
+  if (format.defaultSeats !== undefined) {
+    if (!Number.isFinite(format.defaultSeats)) errors.push('Default seat count must be a number.');
+    if (format.defaultSeats < 2 || format.defaultSeats > 16)
+      errors.push('Default seat count must be between 2 and 16.');
+  }
+
+  for (let i = 0; i < format.packs.length; i++) {
+    const pack = format.packs[i];
+
+    let amount = 0;
+
+    if (!pack.steps) {
+      // this is ok, it just means the pack is a default pack
+      continue;
+    }
+
+    const stepsLength = pack.steps.length;
+    const lastStep = pack.steps[stepsLength - 1];
+    if (lastStep?.action === 'pass') {
+      errors.push(`Pack ${i + 1} cannot end with a pass action.`);
+    }
+
+    for (const step of pack.steps) {
+      if (step === null) {
+        continue;
+      }
+
+      const { action, amount: stepAmount } = step;
+
+      if (action === 'pass') {
+        continue;
+      }
+
+      if (stepAmount !== null) {
+        amount += stepAmount;
+      } else {
+        amount += 1;
+      }
+    }
+
+    if (amount !== pack.slots.length) {
+      errors.push(`Pack ${i + 1} has ${pack.slots.length} slots but has steps to pick or trash ${amount} cards.`);
+    }
+  }
+  return errors.length === 0 ? null : errors;
+};
+
+export const DEFAULT_STEPS: DraftStep[] = [
+  { action: 'pick', amount: 1 },
+  { action: 'pass', amount: null },
+];
+
+export const DEFAULT_PACK: Pack = Object.freeze({ slots: [''], steps: DEFAULT_STEPS });
+
+export const buildDefaultSteps: (cards: number) => DraftStep[] = (cards) => {
+  const steps: DraftStep[] = new Array(cards).fill(DEFAULT_STEPS).flat();
+  // the length should be cards*2-1, because the last pass is removed
+  return normalizeDraftSteps(steps);
+};
+
+export const createDefaultDraftFormat = (packsPerPlayer: number, cardsPerPack: number): DraftFormat => {
+  return {
+    title: `Standard Draft`,
+    packs: Array.from({ length: packsPerPlayer }, () => ({
+      slots: Array.from({ length: cardsPerPack }, () => '*'),
+      steps: buildDefaultSteps(cardsPerPack),
+    })),
+    multiples: false,
+    markdown: '',
+    defaultSeats: 8,
+  };
+};
+
+export const getInitialState = (draft: Draft): State => {
+  const stepQueue: DraftStep[] = [];
+
+  if (draft.InitialState) {
+    // only look at the first seat
+    const seat = draft.InitialState[0];
+
+    for (const pack of seat) {
+      const stepsLength = pack.steps.length;
+      if (stepsLength === 0) {
+        continue;
+      }
+
+      stepQueue.push(...pack.steps);
+
+      //Backwards compatability, add endpack step to the end of the pack if the backend hasn't already
+      if (pack.steps[stepsLength - 1]?.action !== 'endpack') {
```

Comment:
Slight change where endpack is only added if the last step in the pack isn't already that. This made more sense when getInitialState was still in the page component, but now it seems more likely we'd do this in the route and pass the computed initial state to the frontend; if so then this logic isn't that useful.